misk-crypto/src/main/kotlin/misk/crypto/pgp/PgpDecrypterProvider.kt

```diff
@@ -0,0 +1,59 @@
+package misk.crypto.pgp
+
+import com.google.crypto.tink.KmsClient
+import com.google.inject.Inject
+import com.google.inject.Provider
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import misk.crypto.Key
+import misk.crypto.KeyReader
+import misk.moshi.adapter
+import org.bouncycastle.openpgp.PGPSecretKeyRingCollection
+import org.bouncycastle.openpgp.PGPUtil
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder
+import java.util.Base64
+
+internal class PgpDecrypterProvider(
+  private val key: Key,
+  private val kmsUri: String?,
+  private val isReal: Boolean
+) : Provider<PgpDecrypter>, KeyReader() {
+  @Inject private lateinit var kmsClient: KmsClient
+
+  override fun get(): PgpDecrypter {
+    val jsonAdapter = moshi.adapter<PgpKeyJsonFile>()
+    val pgpKeyJsonFile =
+        jsonAdapter.fromJson(key.encrypted_key.value) ?: error("could not deserialize json file")
+
+    val decoded = Base64.getDecoder().decode(pgpKeyJsonFile.encrypted_private_key)
+
+    val secretKeyStream = when (kmsUri) {
```

Comment:
Yeah, this condition shouldn't be here, the provider should assume the KMS URI is never null.
Otherwise, we're introducing here a way to load unencrypted keys which we don't want to.

If this was intended for testing purposes, check out the `FakeKmsClient` class - it should be injected instead of the real client in the tests.